release.sh

```diff
@@ -15,23 +15,25 @@ error () {  # error that writes to stderr, not stdout.
 # SCRIPT_DIR via http://www.ostricher.com/2014/10/the-right-way-to-get-the-directory-of-a-bash-script/
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-# Default variables to empty if not present. Necessary due to the -u option specified above.
-# For more information on this, look here:
-# http://redsymbol.net/articles/unofficial-bash-strict-mode/#solution-positional-parameters
-WORKING_DIR="${1:-}"  # default $1 to empty if it's not supplied
-VERSION="${2:-}"
-OLD_VERSION=   # set later.
-
-# Ensures the current working directory doesn't have tracked but uncommitted files in git.
-clean_working_dir () {
-    if [[ "$(git status -s | grep -m1 "^ ")" ]]; then
-        error "Not checking out release. You have uncommitted files in your working directory."
-        exit 1
-    fi
+# Clone repository into temporary directory
+create_working_dir() {
+    WORKING_DIR=$(mktemp -d)
+    echo "Cloning into working directory at $WORKING_DIR..."
+    cd $WORKING_DIR
+
+    # doing this instead of a git clone so we don't create another directory
+    REPO_URL=$(git -C "$REPO_DIR" remote get-url origin)
+
+    # from http://stackoverflow.com/questions/2411031/how-do-i-clone-into-a-non-empty-directory
+    git init
+    git remote add origin "$REPO_URL"
+    git fetch
+    git checkout -t origin/master
```

Comment:
Yes, it clones into a temp directory and ignores your own checkout of the micromasters repo. Previously there was code which would error if there were files which were not checked out, but this meant you needed to stash those files first, and after the script runs you end up with the git branch being `release-candidate`